### ISSUE TITLE
Line width is doubled

### ISSUE THREAD

**Issue by dessant** [kivy/kivy#3762]
Line.width is doubled, in this example a 50 wide line becomes 100px (it acts the same with smaller widths, like 2 becoming 4px).
I'm not sure if this is known or expected, I haven't seen a reference to it in the docs.

Would it be feasible to move to an implementation with correct width?

```python
from kivy.app import App
from kivy.lang import Builder

kv = """
FloatLayout:
    canvas:
        Color:
            rgba: 1, 1, 1, 1
        Rectangle:
            pos: self.pos
            size: self.size
        Color:
            rgba: 0, 0, 0, 1
        Line:
            points: 100, 100, 100, 400
            width: 50
            cap: 'none'
"""


class TestApp(App):
    def build(self):
        return Builder.load_string(kv)

if __name__ == '__main__':
    TestApp().run()
```

![untitled-4](https://cloud.githubusercontent.com/assets/7513068/11077836/b09dce0a-880a-11e5-9b4d-02a5779c298b.png)

**Comment by inclement** [comment]
This is known, in the sense that I knew about it, but not (as far as I'm aware) in the sense that anyone wrote it down. It comes down to the vertex calculation doubling the width by using it for the offset of vertices from the centre line.

It'd be a breaking change, but maybe we could put it in kivy 2.0.

**Comment by dessant** [comment]
Yes, this always seemed to be the case, we should document it at least.

**Comment by dessant** [comment]
The milestone tag is for the documentation, after that it should be tagged with 2.0.

**Comment by dessant** [comment]
@inclement does the doubling also applies to `width: 1`? I can't decide because of https://github.com/kivy/kivy/issues/3763.

**Comment by inclement** [comment]
I didn't think it did, but your example confused me too, so I'm not sure.

**Comment by janssen** [comment]
Can someone please either fix this or document it?  I'd issue a warning if width is used, also.  That way it will show up in the log, and people will know what's going on.

**Comment by inclement** [comment]
No, with width 1 the line is drawn using an opengl line instruction
instead, which genuinely has width 1 pixel.

**Comment by matham** [comment]
Ideally a module or class variable (depending if this happens in multiple places) called e.g. `WIDTH_DOUBLE` should be added which defaults to True and docs are added. Then, we can make it default to False in a release or two. Assuming we're not ok with just documenting this, not sure why it's double to begin with.

**Comment by Julian-O** [comment]
Confirming it is still happening in Kivy 2.2.1, Windows 11.

I changed the code to:

        points: 100, 100, 100, 200

And I expected a line of length 100, width 50, but got a perfect square.
